crates/ika-core/src/dwallet_mpc/dwallet_mpc_service.rs

```diff
@@ -808,27 +808,19 @@ impl DWalletMPCService {
             "Creating session output message for checkpoint"
         );
         match &session_request.protocol_data {
-            ProtocolData::DWalletDKGWithEncryptedShare {
-                dwallet_id,
-                encrypted_secret_share_id,
-                ..
+            ProtocolData::DWalletDKG {
+                dwallet_id, data, ..
             } => {
                 let tx = DWalletCheckpointMessageKind::RespondDWalletDKGOutput(DWalletDKGOutput {
                     output,
                     dwallet_id: dwallet_id.to_vec(),
-                    encrypted_secret_share_id: Some(encrypted_secret_share_id.to_vec()),
-                    sign_id: None,
-                    signature: vec![],
-                    rejected,
-                    session_sequence_number: session_request.session_sequence_number,
-                });
-                vec![tx]
-            }
-            ProtocolData::DWalletDKGWithPublicShare { dwallet_id, .. } => {
-                let tx = DWalletCheckpointMessageKind::RespondDWalletDKGOutput(DWalletDKGOutput {
-                    output,
-                    dwallet_id: dwallet_id.to_vec(),
-                    encrypted_secret_share_id: None,
+                    encrypted_secret_share_id: match data.user_secret_key_share {
+                        UserSecretKeyShareEventType::Encrypted {
+                            encrypted_user_secret_key_share_id,
+                            ..
+                        } => Some(encrypted_user_secret_key_share_id.to_vec()),
+                        UserSecretKeyShareEventType::Public { .. } => None,
+                    },
```

Comment:
this field is a duplicate now, `data.user_secret_key_share` is sufficient, if its a small change to remove in this pr do it otherwise issue + todo